### PR TITLE
migrating file types

### PR DESCRIPTION
**Pull request by truhacevkir**
- Migrating files
- Adding PendingFile interface
- Updating files to import in brage-import and scopus-import
- Updating logic for AdministrativeAgreement -> PendingInternalFIle/RejectedFile when approving/declining publishing request.

### CODE REVIEW

File: tickets/src/main/java/no/unit/nva/publication/ticket/update/UpdateTicketHandler.java [hunk 1]
```diff
@@ -201,39 +199,32 @@ private void publishingRequestSideEffects(PublishingRequestCase ticket,
...
                                      .build();
         resourceService.updatePublication(updatedPublication);
     }
 
-    private List<AssociatedArtifact> updateUnpublishedFiles(Publication publication) {
+    private List<AssociatedArtifact> rejectFiles(Publication publication) {
         var associatedArtifacts = publication.getAssociatedArtifacts();
         return associatedArtifacts.stream()
-                   .map(this::rejectIfUnpublished)
+                   .map(file -> file instanceof PendingFile<?> pendingFile
```
Comment (brinxmat) [inline tickets/src/main/java/no/unit/nva/publication/ticket/update/UpdateTicketHandler.java:222]: This appears to only reject pending files. Update the name. Also, extract the inlined code here. `rejectPendingFiles`, `rejectSingleFile`

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: tickets/src/main/java/no/unit/nva/publication/ticket/update/UpdateTicketHandler.java (hunk 1 above)
Comment (truhacevkir) [reply tickets/src/main/java/no/unit/nva/publication/ticket/update/UpdateTicketHandler.java]: Moved permanently to AcceptedPublishingRequestEventHandler

File: publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/tickets/AcceptedPublishingRequestEventHandler.java [hunk 2]
```diff
@@ -223,14 +221,10 @@ private List<AssociatedArtifact> publishFilesFromPublishingRequest(
                 .toList();
     }
 
+    //TODO: should be refactored and use PendingFile interface
```
[the remark this answers is not in the document]
Comment (truhacevkir) [reply publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/tickets/AcceptedPublishingRequestEventHandler.java:237]: Refactored it and removed TODO

File: publication-model/src/test/java/no/unit/nva/model/associatedartifacts/file/AssociatedArtifactsMigrationTest.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                         InternalFile.class));
+    }
+
+    public static Stream<Arguments> migratingFilesObjectSupplier() {
+        return Stream.of(Arguments.of(File.builder().buildPublishedFile(), PublishedFile.class, OpenFile.class),
+                         Arguments.of(File.builder().buildUnpublishedFile(), UnpublishedFile.class,
+                                      PendingOpenFile.class),
+                         Arguments.of(File.builder().buildUnpublishableFile(), AdministrativeAgreement.class,
+                                      InternalFile.class));
+    }
```
[the remark this answers is not in the document]
Comment (truhacevkir) [reply publication-model/src/test/java/no/unit/nva/model/associatedartifacts/file/AssociatedArtifactsMigrationTest.java:29]: Adding names

File: tickets/src/main/java/no/unit/nva/publication/ticket/create/TicketResolver.java [hunk 4]
```diff
@@ -232,12 +225,13 @@ private Publication toPublicationWithPublishedFiles(Publication publication) {
...
 
     private List<AssociatedArtifact> convertFilesToPublished(Publication publication) {
-        return publication.getAssociatedArtifacts().stream().map(this::updateFileToPublished).toList();
+        return publication.getAssociatedArtifacts().stream().map(this::openFiles).toList();
     }
 
-    private AssociatedArtifact updateFileToPublished(AssociatedArtifact artifact) {
-        if (isNotAdministrativeAgreement(artifact)) {
-            return ((File) artifact).toPublishedFile();
+    //TODO: Remember to not publish internal files in here
```
[the remark this answers is not in the document]
Comment (truhacevkir) [reply tickets/src/main/java/no/unit/nva/publication/ticket/create/TicketResolver.java]: Fixed, all logic moved to AcceptedPublishingRequestEventHandler

**Pull request merged**